python_modules/dagster/dagster_tests/core_tests/definitions_tests/test_default_io_manager.py

```diff
@@ -85,3 +85,26 @@ def test_asset_override_default_io_manager(instance):
 
         result = execute_job(reconstructable(create_asset_job), instance)
         assert result.success
+
+
+def test_bad_override(instance):
+    with environ(
+        {
+            "DAGSTER_DEFAULT_IO_MANAGER_MODULE": "dagster_tests",
+            "DAGSTER_DEFAULT_IO_MANAGER_ATTRIBUTE": "foo_io_manager_def",
+        }
+    ):
+
+        result = execute_job(reconstructable(fs_io_manager_job), instance)
```

Comment:
could use raise_on_error=False here and check actual exception a la `pytest.raises`